src/WinRPM.jl

```diff
@@ -447,14 +447,23 @@ function do_install(packages::Packages)
     end
 end
 
-const exe7z = iswindows()
-    joinpath(BINDIR, "7z.exe")
-elseif
-    joinpath(BINDIR, "7z")
+const exe7z = if Sys.iswindows()
+    _exe7z = joinpath(Sys.BINDIR, "7z.exe")
+    if isfile(_exe7z)
+        _exe7z
+    else # if it's not located in win-extras lets get it catched by the second isfile check
+        joinpath(Sys.BINDIR, "..", "..", "dist-extras", "7z.exe")
```

Comment:
If the problem is that you're trying to test packages in source builds on windows, it would be better to fix this in your personal build process rather than copy-paste these fallbacks everywhere.

Reply:
So do the stuff for cross compiling and creating an installer!?

Reply:
Or `make install` and copy files around. Otherwise the environment you're testing in and the environment that practically every Windows user will be working in will have discrepancies.

Reply:
Hm, I still don't see the exe in the right folder:
```
sdani@LAPTOP-4CL7JAI3 ~/julia
$ make win-extras
...
sdani@LAPTOP-4CL7JAI3 ~/julia
$ make install

/home/sdani/julia/contrib/install.sh 755 /home/sdani/julia/usr/bin/julia /home/s                                                                                                                                                                                                                                                                                                                                                        dani/julia/julia-abab96668c/bin/
/home/sdani/julia/contrib/install.sh 755 /home/sdani/julia/usr/bin/*.dll /home/s                                                                                                                                                                                                                                                                                                                                                        dani/julia/julia-abab96668c/bin/
/home/sdani/julia/contrib/install.sh 755 /home/sdani/julia/usr/lib/libjulia.dll.                                                                                                                                                                                                                                                                                                                                                        a /home/sdani/julia/julia-abab96668c/lib/
/home/sdani/julia/contrib/install.sh 755 /home/sdani/julia/usr/bin/libopenlibm.d                                                                                                                                                                                                                                                                                                                                                        ll.a /home/sdani/julia/julia-abab96668c/lib/
# Copy public headers
cp -R -L /home/sdani/julia/usr/include/julia/* /home/sdani/julia/julia-abab96668                                                                                                                                                                                                                                                                                                                                                        c/include/julia
# Copy system image
/home/sdani/julia/contrib/install.sh 755 /home/sdani/julia/usr/lib/julia/sys.dll                                                                                                                                                                                                                                                                                                                                                         /home/sdani/julia/julia-abab96668c/lib/julia
# Copy in system image build script
/home/sdani/julia/contrib/install.sh 755 /home/sdani/julia/contrib/build_sysimg.                                                                                                                                                                                                                                                                                                                                                        jl /home/sdani/julia/julia-abab96668c/share/julia/
# Copy in all .jl sources as well
cp -R -L /home/sdani/julia/usr/share/julia /home/sdani/julia/julia-abab96668c/sh                                                                                                                                                                                                                                                                                                                                                        are/
# Copy documentation
cp -R -L /home/sdani/julia/doc/_build/html /home/sdani/julia/julia-abab96668c/sh                                                                                                                                                                                                                                                                                                                                                        are/doc/julia/
# Remove various files which should not be installed
rm -f /home/sdani/julia/julia-abab96668c/share/julia/base/version_git.sh
rm -f /home/sdani/julia/julia-abab96668c/share/julia/test/Makefile
# Copy in beautiful new man page
/home/sdani/julia/contrib/install.sh 644 /home/sdani/julia/usr/share/man/man1/ju                                                                                                                                                                                                                                                                                                                                                        lia.1 /home/sdani/julia/julia-abab96668c/share/man/man1/
# Copy icon and .desktop file
mkdir -p /home/sdani/julia/julia-abab96668c/share/icons/hicolor/scalable/apps/
/home/sdani/julia/contrib/install.sh 644 /home/sdani/julia/contrib/julia.svg /ho                                                                                                                                                                                                                                                                                                                                                        me/sdani/julia/julia-abab96668c/share/icons/hicolor/scalable/apps/
touch -c /home/sdani/julia/julia-abab96668c/share/icons/hicolor/
gtk-update-icon-cache /home/sdani/julia/julia-abab96668c/share/icons/hicolor/
make: gtk-update-icon-cache: Command not found
make: [Makefile:346: install] Error 127 (ignored)
mkdir -p /home/sdani/julia/julia-abab96668c/share/applications/
/home/sdani/julia/contrib/install.sh 644 /home/sdani/julia/contrib/julia.desktop                                                                                                                                                                                                                                                                                                                                                         /home/sdani/julia/julia-abab96668c/share/applications/
# Install appdata file
mkdir -p /home/sdani/julia/julia-abab96668c/share/appdata/
/home/sdani/julia/contrib/install.sh 644 /home/sdani/julia/contrib/julia.appdata                                                                                                                                                                                                                                                                                                                                                        .xml /home/sdani/julia/julia-abab96668c/share/appdata/
# Update RPATH entries and JL_SYSTEM_IMAGE_PATH if ../lib/julia != ../lib/julia
# On FreeBSD, remove the build's libdir from each library's RPATH
mkdir -p /home/sdani/julia/julia-abab96668c/etc
cp -R /home/sdani/julia/usr/etc/julia /home/sdani/julia/julia-abab96668c/etc/

sdani@LAPTOP-4CL7JAI3 ~/julia
$ usr/bin/julia.exe

julia> z7 = joinpath(Sys.BINDIR, "7z.exe")
"C:\\Users\\sdani\\juliastuff\\cygwin\\home\\sdani\\julia\\usr\\bin\\7z.exe"

julia> isfile(z7)
false
```
Not sure whats up with `julia-abab96668c`, but even that bin dir doesn't contain a 7z exe.

I'm not going to start copying random files around :D if make win-extras copies it into this folder, this is good enough for me as long as the normal case is covered first. If you have more patience to fix these fiddly issues, i would very much appreciate if you could just push to this PR.

Reply:
> I'm not going to start copying random files around

Why not? That's what building the installer has always done. Guess it happens as part of `make binary-dist` then. You can locally approximate it. This could be cleaned up in the Julia makefiles, but really isn't that hard to deal with locally - it shouldn't be propagating out into every package that needs to extract files.

Reply:
please remove this line